lib/mix/test/mix/utils_test.exs

```diff
@@ -178,6 +178,53 @@ defmodule Mix.UtilsTest do
     end
   end
 
+  test "verify that write_according_to_opts/3 works as expected" do
+    # ignore any error from this call.
+
+    test_out = "test.out"
+    test_out_bak = "test.out.bak"
+    hello_world = "Hello World!"
+    new_hello_world = "New Hello World!"
+    default_out = "default.out"
+
+    # make sure none of the files exist - ignore errors
+    File.rm(test_out)
+    File.rm(test_out_bak)
+    File.rm(default_out)
+
+    # no optional override - write to the specified default file
+    assert Mix.Utils.write_according_to_opts!(test_out, [hello_world], []) == test_out
```

Comment:
We should not write to the root directory in tests. I suggest to wrap this whole test into our testing helpers, such as:

```
test "foo" do
  in_tmp("write with opts", fn ->
    ...
  end
end
```

This way all writes go to a directory which is already gitignored.